src/pyfmi/fmi3.pyx

```diff
@@ -287,6 +375,189 @@ cdef class FMUModelME3(FMUModelBase3):
+                tolerance,
+                start_time,
+                stop_time_defined,
+                stop_time
+            )
+            self.exit_initialization_mode()
+        except Exception:
+            if not log_open and self.get_log_level() > 2:
+                self._close_log_file()
+
+            raise
+
+        if not log_open and self.get_log_level() > 2:
+            self._close_log_file()
+
+    def enter_initialization_mode(
+        self,
+        tolerance_defined=True,
+        tolerance="Default",
+        start_time="Default",
+        stop_time_defined=False,
+        stop_time="Default"
+    ):
+        """
+        Enters initialization mode by calling the low level FMI function
+        fmi3EnterInitializationMode.
+
+        Note that the method initialize() performs both the enter and
+        exit of initialization mode.
+
+        Args:
+            For a full description of the input arguments, see the docstring for method 'initialize'.
+        """
+        cdef FMIL3.fmi3_status_t status
+
+        cdef FMIL3.fmi3_boolean_t stop_defined = FMIL3.fmi3_true if stop_time_defined else FMIL3.fmi3_false
+        cdef FMIL3.fmi3_boolean_t tol_defined = FMIL3.fmi3_true if tolerance_defined else FMIL3.fmi3_false
+
+        if tolerance == "Default":
+            tolerance = self.get_default_experiment_tolerance()
+        if start_time == "Default":
+            start_time = self.get_default_experiment_start_time()
+        if stop_time == "Default":
+            stop_time = self.get_default_experiment_stop_time()
+
+        self._t = start_time
+        self._last_accepted_time = start_time
+        self._log_handler.capi_start_callback(self._max_log_size_msg_sent, self._current_log_size)
+        status = FMIL3.fmi3_import_enter_initialization_mode(
+            self._fmu,
+            tolerance_defined,
+            tolerance,
+            start_time,
+            stop_time_defined,
+            stop_time
+        )
+        self._log_handler.capi_end_callback(self._max_log_size_msg_sent, self._current_log_size)
+
+        if status != FMIL3.fmi3_status_ok:
+            raise FMUException("Failed to enter initialization mode")
+
+    def exit_initialization_mode(self):
+        """
+            Exit initialization mode by calling the low level FMI function
+            fmi3ExitInitializationMode.
+
+            Note that the method initialize() performs both the enter and
+            exit of initialization mode.
+        """
+        cdef FMIL3.fmi3_status_t status
+        self._log_handler.capi_start_callback(self._max_log_size_msg_sent, self._current_log_size)
+        status = FMIL3.fmi3_import_exit_initialization_mode(self._fmu)
+        self._log_handler.capi_end_callback(self._max_log_size_msg_sent, self._current_log_size)
+
+        if status != FMIL3.fmi3_status_ok:
+            raise FMUException("Failed to exit initialization mode")
+
+    def enter_continuous_time_mode(self):
```

Comment:
This function and `enter_event_mode` I added before I realized I did not need them.